File: src/main/java/org/tindalos/principle/domain/detector/layering/LayerViolationsResult.java
package org.tindalos.principle.domain.detector.layering;

import java.util.List;

import org.tindalos.principle.domain.coredetector.CheckResult;
import org.tindalos.principle.domain.expectations.Layering;

import com.google.common.collect.Lists;

public class LayerViolationsResult implements CheckResult {

    private final List<LayerReference> references;
	private final Layering layeringExpectations;

    public LayerViolationsResult(List<LayerReference> references, Layering layeringExpectations) {
        this.layeringExpectations = layeringExpectations;
		this.references = Lists.newArrayList(references);
    }
    
    public List<LayerReference> getViolations() {
        return references;
    }

	public boolean expectationsFailed() {
		return references.size() > layeringExpectations.getViolationsThreshold();
	}

}

File: src/main/java/org/tindalos/principle/domain/detector/layering/LayerViolationsResult.scala
package org.tindalos.principle.domain.detector.layering

import org.tindalos.principle.domain.coredetector.AnalysisResult
import org.tindalos.principle.domain.expectations.Layering

class LayerViolationsResult(val violations: List[LayerReference], val threshold: Int) extends AnalysisResult {

  override def expectationsFailed() = violations.length > threshold
}